compiler-rt/lib/sanitizer_common/sanitizer_file.cpp

```diff
@@ -19,6 +19,13 @@
 
 #include "sanitizer_common.h"
 #include "sanitizer_file.h"
+
+#if SANITIZER_LINUX || (SANITIZER_ANDROID && __ANDROID_API__ >= 18)
```

Comment:
Actually, I am not sure this works. Wouldn't `SANITIZER_LINUX` also be true for Android? I think we can just drop the Android bit, as ASan wasn't supported at API level 18 anyway.

Reply:
You're right, just found out about `sanitizer_getauxval.h`, so I'll use that definition instead. Is `!SANITIZER_GO` necessary though?  https://github.com/bigb4ng/llvm-project/blob/70efd00055d734e8ca10be3c60ca689906f79924/compiler-rt/lib/sanitizer_common/sanitizer_getauxval.h#L26

Reply:
I'll try to find out if https://github.com/llvm/llvm-project/commit/6c75db8b4bc59eace18143ce086419d37da24746 is still relevant and will let you know as soon as I find out.

Reply:
But what you can do is use the weak definition from that file

`if (getauxval && getauxval(kAtSecure))`

but you'd have to hardcode the `kAtSecure = 23`